src/libs/ReportUtils.ts

```diff
@@ -23,171 +26,193 @@ import isReportMessageAttachment from './isReportMessageAttachment';
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+function isTypeReportAction(arg: ReportAction | Record<string, never>): arg is ReportAction {
+    return arg !== undefined; // Customize this type guard as needed
+}
+
+let currentUserEmail: string | undefined;
+let currentUserAccountID: number | undefined;
+let isAnonymousUser = false;
 
 Onyx.connect({
     key: ONYXKEYS.SESSION,
-    callback: (val) => {
+    callback: (value) => {
         // When signed out, val is undefined
-        if (!val) {
+        if (!value) {
             return;
         }
 
-        currentUserEmail = val.email;
-        currentUserAccountID = val.accountID;
-        isAnonymousUser = val.authTokenType === 'anonymousAccount';
+        currentUserEmail = value.email;
+        currentUserAccountID = value.accountID;
+        isAnonymousUser = value.authTokenType === 'anonymousAccount';
     },
 });
 
-let allPersonalDetails;
-let currentUserPersonalDetails;
+let allPersonalDetails: OnyxCollection<PersonalDetails>;
+let currentUserPersonalDetails: OnyxEntry<PersonalDetails | undefined>;
 Onyx.connect({
     key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-    callback: (val) => {
-        currentUserPersonalDetails = lodashGet(val, currentUserAccountID, {});
-        allPersonalDetails = val || {};
+    callback: (value) => {
+        currentUserPersonalDetails = value?.[currentUserAccountID ?? ''];
+        allPersonalDetails = value ?? {};
     },
 });
 
-let allReports;
+let allReports: OnyxCollection<Report>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT,
     waitForCollectionCallback: true,
     callback: (val) => (allReports = val),
 });
 
-let doesDomainHaveApprovedAccountant;
+let doesDomainHaveApprovedAccountant = false;
 Onyx.connect({
     key: ONYXKEYS.ACCOUNT,
-    waitForCollectionCallback: true,
-    callback: (val) => (doesDomainHaveApprovedAccountant = lodashGet(val, 'doesDomainHaveApprovedAccountant', false)),
+    // waitForCollectionCallback: true,
+    callback: (value) => (doesDomainHaveApprovedAccountant = value?.doesDomainHaveApprovedAccountant ?? false),
 });
 
-let allPolicies;
+let allPolicies: OnyxCollection<Policy>;
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.POLICY,
     waitForCollectionCallback: true,
     callback: (val) => (allPolicies = val),
 });
 
-let loginList;
+let loginList: OnyxEntry<Login>;
 Onyx.connect({
     key: ONYXKEYS.LOGIN_LIST,
     callback: (val) => (loginList = val),
 });
 
-function getChatType(report) {
-    return report ? report.chatType : '';
+function getChatType(report?: OnyxEntry<Report>): ValueOf<typeof CONST.REPORT.CHAT_TYPE> | undefined {
+    return report?.chatType;
 }
 
-/**
- * @param {String} policyID
- * @returns {Object}
- */
-function getPolicy(policyID) {
+function getPolicy(policyID: string): OnyxEntry<Policy> {
     if (!allPolicies || !policyID) {
-        return {};
+        return null;
```

Comment:
@kubabutkiewicz I guess returning null can cause problems here where this value is used https://github.com/Expensify/App/blob/main/src/libs/actions/Policy.js#L219 (policy.name)